pkg/driver/node/credentialprovider/provider_pod.go

```diff
@@ -48,56 +55,68 @@ func (c *Provider) provideFromPod(ctx context.Context, provideCtx ProvideContext
 
 	stsToken := tokens[serviceAccountTokenAudienceSTS]
 	if stsToken == nil {
-		klog.Errorf("credentialprovider: `authenticationSource` configured to `pod` but no service account tokens for %s received. Please make sure to enable `podInfoOnMountCompat`, see "+podLevelCredentialsDocsPage, serviceAccountTokenAudienceSTS)
+		klog.Errorf("credentialprovider: `authenticationSource` configured to `pod` but no service account token for %s received. Please make sure to enable `podInfoOnMountCompat`, see "+podLevelCredentialsDocsPage, serviceAccountTokenAudienceSTS)
 		return nil, status.Errorf(codes.InvalidArgument, "Missing service account token for %s", serviceAccountTokenAudienceSTS)
 	}
 
-	roleARN, err := c.findPodServiceAccountRole(ctx, provideCtx)
-	if err != nil {
-		return nil, err
+	eksToken := tokens[serviceAccountTokenAudiencePodIdentity]
+	if eksToken == nil {
```

Comment:
Just checking: Are we still going to receive Pod Identity token even if Pod Identity Agent is not installed and/or no associations are created? 

In other words, are we always expecting to receive 2 tokens (STS and EKS) in the request?